ethereum/blockcreation/src/main/java/org/hyperledger/besu/ethereum/blockcreation/txselection/BlockTransactionSelector.java

```diff
@@ -358,13 +370,13 @@ public boolean commit() {
     synchronized (isTimeout) {
       isTooLate = isTimeout.get();
       if (!isTooLate) {
+        for (final var pendingAction : selectedTxPendingActions) {
+          pendingAction.run();
+        }
```

Comment:
From what I can see there is no pending action that rely on the state to be committed, so it is fine to running the pending actions before the state commits.

Reply:
Ok thanks